numpy/core/src/umath/_umath_tests.c.src

```diff
@@ -588,11 +588,11 @@ static PyObject *
 UMath_Tests_test_dispatch(PyObject *NPY_UNUSED(dummy), PyObject *NPY_UNUSED(dummy2))
 {
     const char *highest_func, *highest_var;
-    NPY_CPU_DISPATCH_CALL(highest_func = _umath_tests_dispatch_func, ())
-    NPY_CPU_DISPATCH_CALL(highest_var  = _umath_tests_dispatch_var)
+    NPY_CPU_DISPATCH_CALL(highest_func = _umath_tests_dispatch_func, ());
+    NPY_CPU_DISPATCH_CALL(highest_var  = _umath_tests_dispatch_var);
```

Comment:
Note that these could now be rewritten
```suggestion
    highest_func = NPY_CPU_DISPATCH_CALL(_umath_tests_dispatch_func)();
    highest_var = NPY_CPU_DISPATCH_CALL(_umath_tests_dispatch_var);
```
Since this is a test, I decided to leave them as they are for now.

Reply:
> The main motivation here is that expressions are great, and making the result of the call an expression means I can use it like any other function call.

I will always vote for flexibility/hackability especially when you don't know "how deep the rabbit hole goes".
I see another thing here that may be reducing the flexibility will force the contributors towards safety, and I guess
this patch wouldn't harm that much we can always add new macros to serve different needs or even handle it locally in the case of #16960.

> those were flexible enough that I could just change the NPY_CPU one

We can speed up the multiple calls of NPY_CPU_HAVE by inlining `npy_cpu_have` and extern the local variable `npy__cpu_have` which will give a good hint to the compiler without the need to change the NPY_CPU(CHK) one. 

https://github.com/numpy/numpy/blob/60f9c616aa12800e27baef1db0f964cd9da6623f/numpy/core/src/common/npy_cpu_features.h#L111-L115

https://github.com/numpy/numpy/blob/60f9c616aa12800e27baef1db0f964cd9da6623f/numpy/core/src/common/npy_cpu_features.c.src#L9

Reply:
You can just change one of them so we can test both behaviors.